command/agent/config_test.go

```diff
@@ -114,6 +109,11 @@ func TestConfig_Merge(t *testing.T) {
 			NumSchedulers:     2,
 			EnabledSchedulers: []string{structs.JobTypeBatch},
 			NodeGCThreshold:   "12h",
+			RejoinAfterLeave:  true,
```

Comment:
> Is there a specific test case that you are thinking of?

No, I just like to make sure we test the default case where none of the new options is specified so we don't get a regression at some point. If there is not an existing test case that covers this we should add a new one. I asked because it's hard to see this from the diff; I will look more closely later.